src/code/PreRender.c

```diff
@@ -1,42 +1,632 @@
+                        G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
+                    G_AC_NONE | G_ZS_PRIM | G_RM_PASS | G_RM_OPA_CI2);
+    gDPSetCombineLERP(dListHead++, 0, 0, 0, TEXEL0, 0, 0, 0, 0, 0, 0, 0, TEXEL0, 0, 0, 0, 0);
+    gDPSetColorImage(dListHead++, G_IM_FMT_I, G_IM_SIZ_8b, this->width, cvgSave);
+    gDPSetScissor(dListHead++, G_SC_NON_INTERLACE, 0, 0, this->width, this->height);
+
+    x = this->height;
+    x2 = 0;
+    add = 0x1000 / (this->width * 2);
+
+    while (x > 0) {
+        lrs = this->width - 1;
+        if (x < add)
+            add = x;
+        uls = 0;
+        ult = x2;
+        lrt = (x2 + add) - 1;
+
+        gDPLoadTextureTile(dListHead++, fbuf, G_IM_FMT_IA, G_IM_SIZ_16b, this->width, this->height, uls, ult, lrs, lrt,
+                           0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                           G_TX_NOLOD, G_TX_NOLOD);
+        if (1) {}
+        gSPTextureRectangle(dListHead++, uls << 2, ult << 2, (lrs + 1) << 2, (lrt + 1) << 2, G_TX_RENDERTILE, uls << 5,
+                            ult << 5, 1 << 10, 1 << 10);
+
+        x -= add;
+        x2 += add;
+    }
+
+    gDPPipeSync(dListHead++);
+    gDPSetColorImage(dListHead++, G_IM_FMT_RGBA, G_IM_SIZ_16b, this->width, this->fbuf);
+    *dList = dListHead;
+}
+
+void func_800C1E9C(PreRenderContext* this, Gfx** dList) {
+    LogUtils_CheckNullPointer("this->zbuf_save", this->zbufSave, "../PreRender.c", 481);
+    LogUtils_CheckNullPointer("this->zbuf", this->zbuf, "../PreRender.c", 482);
+
+    if (!this->zbufSave || !this->zbuf) {
+        return;
+    }
+
+    func_800C0F28(this, dList, this->zbuf, this->zbufSave);
+}
+
+void func_800C1F20(PreRenderContext* this, Gfx** dList) {
+    LogUtils_CheckNullPointer("this->fbuf_save", this->fbufSave, "../PreRender.c", 495);
+    LogUtils_CheckNullPointer("this->fbuf", this->fbuf, "../PreRender.c", 496);
+
+    if (!this->fbufSave || !this->fbuf) {
+        return;
+    }
+
+    func_800C1AE8(this, dList, this->fbuf, this->fbufSave);
+}
+
+void func_800C1FA4(PreRenderContext* this, Gfx** dList) {
+    Gfx* dListHead;
+
+    dListHead = *dList;
+
+    gDPPipeSync(dListHead++);
+    gDPSetBlendColor(dListHead++, 0xFF, 0xFF, 0xFF, 0x08);
+    gDPSetPrimDepth(dListHead++, -1, -1);
+    gDPSetOtherMode(dListHead++,
+                    G_AD_DISABLE | G_CD_DISABLE | G_CK_NONE | G_TC_FILT | G_TF_POINT | G_TT_NONE | G_TL_TILE |
+                        G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
+                    G_AC_NONE | G_ZS_PRIM | G_RM_VISCVG | G_RM_VISCVG2);
+    gDPSetScissor(dListHead++, G_SC_NON_INTERLACE, 0, 0, this->width, this->height);
+    gDPFillRectangle(dListHead++, 0, 0, this->width, this->height);
+    gDPPipeSync(dListHead++);
+
+    *dList = dListHead;
+}
+
+void func_800C20B4(PreRenderContext* this, Gfx** dList) {
```

Comment:
It looks like this function and other "draw" entrypoints of this system are called with a pointer to the head of a display buffer, not a dlist. Based on our previous discussion, we would call that `dispHeadp` or `dispHeadP`.

However, since this system doesn't know that fact, and it could very well work with a pointer the the head of a dlist, I think we should just have it as a generic `gfxp`, which becomes `gfx` when dereferencing it later on in other functions. Based on various strings in this file, they even had them as `glistpp`/`glistp`, which likely means a generic "pointer to gfx list" (or pointer to pointer for `pp`), and confirms the generic intent here.